File: 419-battleships-in-a-board/419-battleships-in-a-board.cpp
class Solution {
public:
    int countBattleships(vector<vector<char>>& board) {
       int rows = board.size(), cols = board[0].size(), count = 0;
        for(int idx = 0; idx < rows; idx++){
            for(int idy = 0; idy <cols;idy++){
                if(board[idx][idy]=='X'){
                    if(idx-1 >= 0 && board[idx-1][idy]=='X')
                        continue;
                    if(idy-1 >= 0 && board[idx][idy-1]=='X')
                        continue;
                    count++;
                }
            }
        }
        return count;
    }
};

File: 419-battleships-in-a-board/419-battleships-in-a-board.py
class Solution:
    def countBattleships(self, board: List[List[str]]) -> int:
        n = len(board)
        m = len(board[0])
        count = 0
        for i in range(n):
            for j in range(m):
                if board[i][j]=='X':
                    if ( i > 0 and board[i-1][j]=='X' )or  (j  > 0 and board[i][j-1] =='X'):
                        continue
                    count += 1
        return count
        